### DISCUSSION TITLE
Vercel mixing up client & server endpoints and calling strange urls

### DISCUSSION THREAD

**Discussion by MuhammadAli13562** [vercel/community#6520] [Help]
### Summary

I have my react + vite app at : 
`https://chat-app-sigma-two-12.vercel.app/`

I have my server at domain :
`https://chatapp.ali-98-ec2-backend.click`

I have added domain as a string in envirnment variable in vercel envirnment variables.

But now when i go to sign-in page at  `https://chat-app-sigma-two-12.vercel.app/sign-in `  vercel is calling some strange urls.

Expected URL called from client:
`https://chatapp.ali-98-ec2-backend.click/auth/verify`

Actual URL called from client:
`https://chat-app-sigma-two-12.vercel.app/%22https:/chatapp.ali-98-ec2-backend.click/%22/auth/verify`

I dont know why it is mixing up in this bizzare way.

### Steps to Reproduce

Go to  `https://chat-app-sigma-two-12.vercel.app/`
Reload & Look at console the client & server url is mixed up and called

**Comment by versecafe** [comment]
@MuhammadAli13562 It appears to be calling your base URL + a URI safe version of the target URL leading to the strange URL being generated do you have something like a Link element that is using an href of `/+${targetUrl}` instead of just the targetUrl?

**Reply by MuhammadAli13562** [reply]
Here is how the env variable is flowing in codebase:

First Importing the env variale:

`export const ServerEndpoint = import.meta.env.VITE_SERVER_ENDPOINT`

Creating Redux Toolkit  Api

```
export const api = createApi({
  baseQuery: fetchBaseQuery({ baseUrl: ServerEndpoint }),
  endpoints: () => ({}),
})

```
THEN INTO AUTH ENDPOINT RTK QUERIES
```
const AuthApi = api.injectEndpoints({
  endpoints: builder => ({
    SignInUser: builder.mutation<SignInReturns, SignInCreds>({
      query: (creds: SignInCreds) => ({
        method: "post",
        url: "/auth/login",
        headers: {
          ...creds,
        },
      }),
    }),
    VerifyUserLogin: builder.mutation<Response, void>({
      query: () => ({
        method: "post",
        url: "/auth/verify",
        headers: {
          token: localStorage.getItem("token") || "",
        },
      }),
    }),
  }),
  overrideExisting: false,
})

```

here is the envirnment variable inserted into vercel settings:

Name:
`VITE_SERVER_ENDPOINT`

Value:
` "https://chatapp.ali-98-ec2-backend.click/" `
